app/helpers/application_helper.rb

```diff
@@ -79,4 +79,25 @@ def user_access_options(access_options = %w[true false])
       OpenStruct.new(label: t("users.has_access.#{access}.name"), value: access, description: t("users.has_access.#{access}.description"))
     end
   end
+
+  def sign_in_button(is_e2e_user:)
+    govuk_button_to t("sign_in_button"), omniauth_authorize_path, params: sign_in_params(is_e2e_user:), data: { module: "sign-in-button" }
+  end
+
+  def sign_up_button(is_e2e_user:)
+    govuk_button_to t("sign_up_button"), omniauth_authorize_path, params: sign_in_params(is_e2e_user:, login_type: :sign_up), data: { module: "sign-in-button" }
+  end
+
+  def omniauth_authorize_path
+    "/auth/#{Settings.auth_provider}/"
+  end
```

Comment:
This won't work for Signon (see definition of `default_provider` method in AuthenticationController), but I'm guessing you knew that already.

Reply:
I didn't! I guess it won't matter if we are removing signon in the near future anyway?

Reply:
Yeah I'd reckon so.